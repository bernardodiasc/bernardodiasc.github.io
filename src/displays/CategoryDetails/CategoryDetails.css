.CategoryDetails {
  font-size: var(--text-size);
  margin: 0 0 2em;
}
