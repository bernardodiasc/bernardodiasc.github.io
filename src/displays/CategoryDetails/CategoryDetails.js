import React, { PureComponent } from 'react'
import marked from 'marked'
import renderHTML from 'react-render-html'
import './CategoryDetails.css'

import TextBlock from 'displays/TextBlock'

class CategoryDetails extends PureComponent {
  static defaultProps = {
    category: {
      title: '',
      icon: '',
      body: '',
    },
  }

  render() {
    const { title, body } = this.props.category
    return (
      <div className="CategoryDetails">
        <TextBlock>
          <h1><small>Category:</small><br />{title}</h1>
          {body && renderHTML(marked(body))}
          <hr />
        </TextBlock>
      </div>
    )
  }
}

export default CategoryDetails
