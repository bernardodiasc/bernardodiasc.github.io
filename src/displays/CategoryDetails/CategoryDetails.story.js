import React from 'react'
import { storiesOf } from '@storybook/react'
import { withKnobs } from '@storybook/addon-knobs/react'
import StoryRouter from 'storybook-router'
import Provider from 'stories/containers/Provider'
import Docs from 'stories/containers/Docs'

import CategoryDetails from './CategoryDetails'
import CategoryDetailsReadme from './README.md'
storiesOf('Displays/CategoryDetails', module)
  .addDecorator(story => <Provider story={story()} />)
  .addDecorator(StoryRouter())
  .addDecorator(withKnobs)
  .add('Info', () => <Docs>{CategoryDetailsReadme}</Docs>)
  .add('Example', () => <CategoryDetails />)
  .add('Minimal render', () => <CategoryDetails />)
