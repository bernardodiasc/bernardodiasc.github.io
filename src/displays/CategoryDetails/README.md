# CategoryDetails

## How to use

```
import CategoryDetails from 'displays/CategoryDetails'
```

```
<CategoryDetails />
```

## Properties

| propName | propType | defaultValue | isRequired |
| - | - | - | - |
