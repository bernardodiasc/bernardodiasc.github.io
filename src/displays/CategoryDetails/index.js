import CategoryDetails from './CategoryDetails'
export default CategoryDetails
