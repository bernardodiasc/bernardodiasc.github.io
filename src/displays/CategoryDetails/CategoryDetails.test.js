import React from 'react'
import { shallow } from 'enzyme'

import CategoryDetails from './CategoryDetails.js'

it('renders without crashing', () => {
  const component = shallow(<CategoryDetails />)
  expect(component).toMatchSnapshot()
})
