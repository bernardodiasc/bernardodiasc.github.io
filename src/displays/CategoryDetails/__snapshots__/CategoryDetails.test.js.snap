// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="CategoryDetails"
>
  <TextBlock>
    <h1>
      <small>
        Category:
      </small>
      <br />
    </h1>
    <hr />
  </TextBlock>
</div>
`;
